backend/inspirehep/snow/api.py

```diff
@@ -0,0 +1,571 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+import urllib.parse
+
+import orjson
+import requests
+import structlog
+from flask import current_app, render_template
+from inspire_utils.record import get_value
+
+from .errors import CreateTicketException, EditTicketException, SnowAuthenticationError
+from .proxies import snow_instance
+from .utils import get_from_cache, get_response_result, relogin_if_needed, strip_lines
+
+LOGGER = structlog.getLogger()
+
+
+class SnowTicketAPI:
+    """Implements basic methods to interact with SNOW (CERN implementation)"""
+
+    def __init__(self, url, client_id, secret, auth_url, verify_cert):
+        self.client_id = client_id
+        self.secret = secret
+        self.auth_url = auth_url
+        self.verify_cert = verify_cert
+        self.url = url
+
+    @property
+    def headers(self):
+        return {
+            "Authorization": f"Bearer {self.token}",
+            "Content-Type": "application/json",
+        }
+
+    def login(self):
+        login_payload = {
+            "grant_type": "client_credentials",
+            "audience": "snow-api-sso-protected",
+            "client_id": self.client_id,
+            "client_secret": self.secret,
+        }
+        headers = {
+            "Content-Type": "application/x-www-form-urlencoded",
+        }
+        try:
+            response = requests.post(self.auth_url, data=login_payload, headers=headers)
+            response.raise_for_status()
+            self.token = response.json()["access_token"]
```

Comment:
do we know for how long we keep the token? do we have to send the request to get it? Couldn't we just set it on our configmaps?

Reply:
If that's possible, it's much easier, we can just drop all the flask ext/proxy

Reply:
token is valid for 20 min, we have to send a request to get it

Reply:
> If that's possible, it's much easier, we can just drop all the flask ext/proxy

I think we can, I thought it was our coding convention 😉